develop/dev-guide-sample-application-java-hibernate.md

```diff
@@ -1,374 +1,173 @@
+    - 参考[部署本地测试 TiDB 集群](/quick-start-with-tidb.md#部署本地测试集群)或[部署正式 TiDB 集群](/production-deployment-using-tiup.md)，创建本地集群。
 
-## 第 1 步：启动你的 TiDB 集群
+## 运行代码并连接到 TiDB
 
-本节将介绍 TiDB 集群的启动方法。
+本小节演示如何运行示例应用程序的代码，并连接到 TiDB。
 
-**使用 TiDB Serverless 集群**
+### 第 1 步：克隆示例代码仓库到本地
 
-详细步骤，请参考：[创建 TiDB Serverless 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-tidb-serverless-集群)。
+在命令行中运行以下命令，克隆示例代码仓库到本地：
 
-**使用本地集群**
+```bash
+git clone https://github.com/tidb-samples/tidb-java-hibernate-quickstart.git
+cd tidb-java-hibernate-quickstart
+```
 
-详细步骤，请参考：[部署本地测试 TiDB 集群](/quick-start-with-tidb.md#部署本地测试集群)或[部署正式 TiDB 集群](/production-deployment-using-tiup.md)。
+### 第 2 步：配置连接信息
 
-## 第 2 步：获取代码
+根据不同的 TiDB 部署方式，使用不同的方法连接到 TiDB 集群。
 
-```shell
-git clone https://github.com/pingcap-inc/tidb-example-java.git
-```
+<SimpleTab>
 
-与 [Hibernate](https://hibernate.org/orm/) 对比，JDBC 的实现方式并非最优体验。你需要自行编写错误处理逻辑，并且代码无法简单复用。这会使你的代码有些冗余。
+<div label="TiDB Serverless">
 
-此处将以 `6.0.0.Beta2` 版本进行说明。
+1. 在 TiDB Cloud Web Console 中，选择你的 TiDB Serverless 集群，进入 **Overview** 页面。
 
-进入目录 `plain-java-hibernate`：
+2. 点击右上角的 **Connect** 按钮，会显示连接对话框。
 
-```shell
-cd plain-java-hibernate
-```
+3. 确认窗口中的配置和你的运行环境一致。
 
-目录结构如下所示：
+    - **Endpoint Type** 为 `Public`
+    - **Connect With** 选择 `General`
+    - **Operating System** 为你的运行环境
 
-```
-.
-├── Makefile
-├── plain-java-hibernate.iml
-├── pom.xml
-└── src
-    └── main
-        ├── java
-        │   └── com
-        │       └── pingcap
-        │           └── HibernateExample.java
-        └── resources
-            └── hibernate.cfg.xml
-```
+    <Tip>如果你在 Windows Subsystem for Linux (WSL) 中运行，请切换为对应的 Linux 发行版。</Tip>
 
-其中，`hibernate.cfg.xml` 为 Hibernate 配置文件，定义了：
+4. 点击 **Create password** 生成密码。
 
-```xml
-<?xml version='1.0' encoding='utf-8'?>
-<!DOCTYPE hibernate-configuration PUBLIC
-        "-//Hibernate/Hibernate Configuration DTD 3.0//EN"
-        "http://www.hibernate.org/dtd/hibernate-configuration-3.0.dtd">
-<hibernate-configuration>
-    <session-factory>
+   <Tip>如果你之前已经生成过密码，可以直接使用原密码，或点击 **Reset password** 重新生成密码。</Tip>
 
-        <!-- Database connection settings -->
-        <property name="hibernate.connection.driver_class">com.mysql.cj.jdbc.Driver</property>
-        <property name="hibernate.dialect">org.hibernate.dialect.TiDBDialect</property>
-        <property name="hibernate.connection.url">jdbc:mysql://localhost:4000/test</property>
-        <property name="hibernate.connection.username">root</property>
-        <property name="hibernate.connection.password"></property>
-        <property name="hibernate.connection.autocommit">false</property>
+5. 运行以下命令，将 `env.sh.example` 复制并重命名为 `env.sh`：
 
-        <!-- Required so a table can be created from the 'PlayerDAO' class -->
-        <property name="hibernate.hbm2ddl.auto">create-drop</property>
+    ```bash
+    cp env.sh.example env.sh
```

Comment:
The `env.sh` script always use SSL in JDBC URL.

You need to add spaces at both ends of the equal sign `=` in the if statement:

```shell
if [ 'true' = "${USE_SSL}" ]; then
  jdbc_url="${jdbc_url}?sslMode=VERIFY_IDENTITY&enabledTLSProtocols=TLSv1.2,TLSv1.3"
fi
```

https://stackoverflow.com/questions/4977367/why-equal-to-operator-does-not-work-if-it-is-not-surrounded-by-space

Reply:
You are right! Good catch!